xds/internal/balancer/edsbalancer/edsbalancer_test.go

```diff
@@ -364,6 +379,59 @@ func TestEDS_TwoLocalities(t *testing.T) {
 	}
 }
 
+// The EDS balancer gets EDS resp with unhealthy endpoints. Test that only
+// healthy ones are used.
+func TestEDS_EndpointsHealth(t *testing.T) {
+	cc := newTestClientConn(t)
+	edsb := NewXDSBalancer(cc, nil)
+
+	// Two localities, each 3 backend, one Healthy, one Unhealthy, one Unknown.
+	clab1 := newClusterLoadAssignmentBuilder(testClusterNames[0], nil)
+	clab1.addLocality(testSubZones[0], 1, testEndpointAddrs[:3], &addLocalityOptions{
+		health: []corepb.HealthStatus{
+			corepb.HealthStatus_HEALTHY,
+			corepb.HealthStatus_UNHEALTHY,
```

Comment:
Looks like there is also DRAINING, TIMEOUT, and DEGRADED.  We should ideally include those as well.

Reply:
Done